tests/test_series.py

```diff
@@ -1330,3 +1330,12 @@ def test_logical_operators() -> None:
         pd.Series,
         bool,
     )
+
+
+def test_AnyArrayLike_and_clip() -> None:
+    ser = pd.Series([1, 2, 3])
+    s1=ser.clip(upper=ser)
+    s2=ser.clip(upper=ser)
+    check(assert_type(s1, pd.Series), pd.Series, pd.Series[Any])
+    check(assert_type(s2, pd.Series), pd.Series, pd.Series[Any])
```

Comment:
`check(assert_type(s2, pd.Series), pd.Series)`

If you know the type of the object inside the `Series`, then you do `check(assert_type(s2, pd.Series), pd.Series, float)` (for example).  But here we don't know the type, so we just leave out that last argument to `check`. 

Look at the code for `check` to see how it works